### PR TITLE
feat(nat): private nat gateway datasource support tags filter

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

Private nat gateway datasource support tags filter

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1.Add parameter tags as filtering condition
2.Modify the non-standard name 
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
root@ecs-ruwenqiang:/home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud (br_privatenatlist)$ make testacc TEST="./huaweicloud/services/acceptance/nat" TESTARGS="-run TestAccDatasourcePrivateGateways_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/nat -v -run TestAccDatasourcePrivateGateways_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourcePrivateGateways_basic
=== PAUSE TestAccDatasourcePrivateGateways_basic
=== CONT  TestAccDatasourcePrivateGateways_basic
--- PASS: TestAccDatasourcePrivateGateways_basic (112.63s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/nat       112.684s
```

### CODE REVIEW

File: docs/data-sources/nat_private_gateways.md [hunk 1]
```diff
@@ -20,14 +20,14 @@ data "huaweicloud_nat_private_gateways" "test" {
 
 The following arguments are supported:
 
-* `region` - (Optional, String) Specifies the region where the private NAT gateways are located.  
+* `region` - (Optional, String) Specifies the region where the private NAT gateways are located.
```
Comment (Lance52259) [inline docs/data-sources/nat_private_gateways.md]:
```suggestion
* `region` - (Optional, String) Specifies the region where the private NAT gateways are located.  
```
two spaces at the end of the each line means force break.

File: huaweicloud/services/nat/data_source_huaweicloud_nat_private_gateways.go [hunk 2]
```diff
@@ -213,16 +220,23 @@ func flattenListGatewaysResponseBodyGateways(resp interface{}) []interface{} {
 			"vpc_id":                utils.PathSearch("downlink_vpcs[0].vpc_id", v, nil),
 			"subnet_id":             utils.PathSearch("downlink_vpcs[0].virsubnet_id", v, nil),
 			"enterprise_project_id": utils.PathSearch("enterprise_project_id", v, nil),
+			"tags":                  utils.FlattenTagsToMap(utils.PathSearch("tags", v, nil)),
 		})
 	}
 	return rst
 }
 
 func filterListGatewaysResponseBodyGateways(all []interface{}, d *schema.ResourceData) []interface{} {
```
Comment (Lance52259) [inline huaweicloud/services/nat/data_source_huaweicloud_nat_private_gateways.go]:
```suggestion
func filterListGatewaysResponseByTags(all []interface{}, d *schema.ResourceData) []interface{} {
```
This function is used only to filter results by tags. The current name may cause ambiguity.

File: docs/data-sources/nat_private_gateways.md (hunk 1 above)
Comment (ruwenqiang123) [reply docs/data-sources/nat_private_gateways.md]: done

File: huaweicloud/services/nat/data_source_huaweicloud_nat_private_gateways.go (hunk 2 above)
Comment (ruwenqiang123) [reply huaweicloud/services/nat/data_source_huaweicloud_nat_private_gateways.go]: done

**Pull request merged**